databricks_cli/stack/api.py

```diff
@@ -0,0 +1,362 @@
+            resource_map_key = (resource_config.get(RESOURCE_ID),
+                                resource_config.get(RESOURCE_SERVICE))
+            resource_status = resource_id_to_status.get(resource_map_key) \
+                if resource_map_key in resource_id_to_status else None
+            # Deploy resource, get resource_status
+            new_resource_status = self._deploy_resource(resource_config, resource_status)
+            resource_statuses.append(new_resource_status)
+
+        # stack deploy status is original config with deployed resource statuses added
+        new_stack_status = copy.deepcopy(stack_config)
+        new_stack_status.update({STACK_DEPLOYED: resource_statuses})
+        new_stack_status.update({CLI_VERSION_KEY: CLI_VERSION})
+
+        # Validate that the status has been created correctly
+        self._validate_status(new_stack_status)
+
+        return new_stack_status
+
+    def _deploy_resource(self, resource_config, resource_status=None):  # overwrite to be added
+        """
+        Deploys a resource given a resource information extracted from the stack JSON configuration
+        template.
+
+        :param resource_config: A dict of the resource with fields of RESOURCE_ID, RESOURCE_SERVICE
+        and RESOURCE_PROPERTIES.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'properties': {...}}
+        :param resource_status: A dict of the resource's deployment info from the last
+        deployment. Will be None if this is the first deployment.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'physical_id': {...}}
+        :return: dict resource_status- A dictionary of deployment information of the
+        resource to be stored at deploy time. It includes the resource id of the resource along
+        with the physical id and deploy output of the resource.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'physical_id': {'job_id': 123},
+        'timestamp': 123456789, 'deploy_output': {..}}
+        """
+        resource_id = resource_config.get(RESOURCE_ID)
+        resource_service = resource_config.get(RESOURCE_SERVICE)
+        resource_properties = resource_config.get(RESOURCE_PROPERTIES)
+        physical_id = resource_status.get(RESOURCE_PHYSICAL_ID) if resource_status else None
+
+        if resource_service == JOBS_SERVICE:
+            click.echo("Deploying job '{}' with properties: \n{}".format(resource_id, json.dumps(
+                resource_properties, indent=2, separators=(',', ': '))))
+            new_physical_id, deploy_output = self._deploy_job(resource_properties,
+                                                              physical_id)
+        else:
+            raise StackError("Resource service '{}' not supported".format(resource_service))
+
+        new_resource_status = {RESOURCE_ID: resource_id, RESOURCE_SERVICE: resource_service,
+                               RESOURCE_DEPLOY_TIMESTAMP: datetime.now(),
```

Comment:
why bother with the conversion in `_json_type_handler`? Can we just leave everything as a milliseconds after epoch?